library/model/book.py

```diff
@@ -54,38 +54,19 @@ def from_borrowed_book(cls, borrowed_book: "BorrowedBook") -> "Book":
         return book
 
     def can_borrow(self) -> bool:
-        if self._book_type == "Paper":
-            return self.existing_items - self.borrowed_items > 0
-        elif self._book_type == "Electronic":
-            return True
-        elif self._book_type == "Audio":
-            return True
-        else:
-            raise AttributeError("No such book type...")
+            return BookTypeProcessor.can_borrow(self)
+        
 
     def get_approximate_duration(self) -> int:
-        if self._book_type == "Paper":
-            return self.pages * 3 * 60
-        elif self._book_type == "Electronic":
-            return self.pages * 5 * 60
-        elif self._book_type == "Audio":
-            return self.duration
-        else:
-            raise AttributeError("No such book type...")
+            return BookTypeProcessor.get_approximate_duration(self)
+        
 
     def get_weekly_fee(self) -> int:
-        if self._book_type == "Paper":
-            return 5
-        elif self._book_type == "Electronic":
-            return 2
-        elif self._book_type == "Audio":
-            return 2
-        else:
-            raise AttributeError("No such book type...")
-
+        return BookTypeProcessor.get_weekly_fee(self)
+        
     def borrow_book(self) -> "BorrowedBook":
         if self.can_borrow():
-            if self._book_type == "Paper":
+            if BookTypeProcessor.book_check_reqrd(self):
```

Comment:
In my opinion, the readability of the if statement decreased here. You could follow the prefix pattern (`is` `should` `need` etc) for methods returning a bool.